contracts/protocol/integration/wrap-v2/ERC4626WrapV2Adapter.sol

```diff
@@ -0,0 +1,111 @@
+import { IERC4626 } from "../../../interfaces/external/IERC4626.sol";
+
+/**
+ * @title ERC4626WrapV2Adapter
+ * @author pblivin0x
+ *
+ * Wrap adapter for ERC-4626 Vaults that returns data for wraps/unwraps of tokens
+ */
+contract ERC4626WrapV2Adapter {
+
+    /* ============ External Getter Functions ============ */
+
+    /**
+     * Generates the calldata to wrap an underlying asset into a wrappedToken.
+     *
+     * @param _underlyingToken      Address of the component to be wrapped
+     * @param _wrappedToken         Address of the desired wrapped token
+     * @param _underlyingUnits      Total quantity of underlying units to wrap
+     * @param _to                   Address to send the wrapped tokens to
+     *
+     * @return address              Target contract address
+     * @return uint256              Total quantity of underlying units (if underlying is ETH)
+     * @return bytes                Wrap calldata
+     */
+    function getWrapCallData(
+        address _underlyingToken,
+        address _wrappedToken,
+        uint256 _underlyingUnits,
+        address _to,
+        bytes memory /* _wrapData */
+    )
+        external
+        view
+        returns (address, uint256, bytes memory)
+    {
+        IERC4626 vault = IERC4626(_wrappedToken);
+        require(vault.asset() == _underlyingToken, "Must be a valid token pair");
+
+        bytes memory callData = abi.encodeWithSignature(
+            "deposit(uint256,address)",
+            _underlyingUnits,
+            _to
+        );
+
+        return (address(vault), 0, callData);
+    }
+
+    /**
+     * Generates the calldata to unwrap a wrapped asset into its underlying.
+     *
+     * @param _underlyingToken      Address of the underlying asset
+     * @param _wrappedToken         Address of the component to be unwrapped
+     * @param _wrappedTokenUnits    Total quantity of wrapped token units to unwrap
+     * @param _to                   Address to send the unwrapped tokens to
+     *
+     * @return address              Target contract address
+     * @return uint256              Total quantity of wrapped token units to unwrap. This will always be 0 for unwrapping
+     * @return bytes                Unwrap calldata
+     */
+    function getUnwrapCallData(
+        address _underlyingToken,
+        address _wrappedToken,
+        uint256 _wrappedTokenUnits,
+        address _to,
+        bytes memory /* _wrapData */
+    )
+        external
+        view
+        returns (address, uint256, bytes memory)
+    {
+        IERC4626 vault = IERC4626(_wrappedToken);
+        require(vault.asset() == _underlyingToken, "Must be a valid token pair");
+
+        bytes memory callData = abi.encodeWithSignature(
+            "withdraw(uint256,address,address)",
+            _wrappedTokenUnits,
+            _to,
+            _to
+        );
+
+        return (address(vault), 0, callData);
+    }
+
+    /**
+     * Returns the address to approve source tokens for wrapping.
```

Comment:
Nit: missing param documentation. (Probably bot really necessary but might add just for consistency)

Reply:
Added here [c6464bd](https://github.com/IndexCoop/index-protocol/pull/11/commits/c6464bdd5b42f566553785b6acb29d156869a1e8)